### ISSUE TITLE
Update `cloud.Bucket` sim implementation to use `sim.Container` with MinIO

### ISSUE THREAD

**Issue by Chriscbr** [winglang/wing#6300]
### Use Case

`cloud.Bucket` is an abstract object storage API which is designed to be cloud agnostic (the API's design prioritizes supporting APIs that are available in multiple cloud provider services, like AWS S3, Google Cloud Storage, and Azure Blob Storage). Currently the Wing SDK uses a hand-written object storage implementation found here: https://github.com/winglang/wing/blob/main/libs/wingsdk/src/target-sim/bucket.inflight.ts.

As an alternative, we could implement the API by running a docker container running an existing object storage implementation, like [MinIO](https://min.io/docs/minio/linux/index.html). The SDK has existing support for this via `sim.Container`.

Pros:
* No more need to maintain a hand-written object storage implementation - our existing [AWS S3 client bindings](https://github.com/winglang/wing/blob/main/libs/wingsdk/src/shared-aws/bucket.inflight.ts) can be reused
* It could make it easier to add support for additional S3 features, like object tagging, or using the bucket to host static website files.

Cons:
* Using a container could impose higher CPU/memory constraints.
* Using an S3-compatible backing implementation could make it easier to violate our tenet of avoiding cloud bias when designing `cloud` APIs.

### Implementation Notes

Related: https://github.com/winglang/wing/issues/6282

List of S3 APIs supported by MinIO: https://min.io/docs/minio/linux/reference/s3-api-compatibility.html

### Component

SDK

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.
* If this issue is labeled **needs-discussion**, it means the spec has not been finalized yet. Please reach out on the #dev channel in the [Wing Slack](https://t.winglang.io/slack).

**Comment by garysassano** [comment]
Makes sense, and in line with what I wrote [here](https://github.com/winglang/wing/issues/6282#issuecomment-2068126152).

MinIO is basically S3 compiled for `sim` platform, which is easily portable to many cloud providers.

**Comment by Chriscbr** [comment]
> Do we actually need to change the `cloud.Bucket` implementation?
> 
> For how I see it, MinIO is simply the [s3 winglib](https://github.com/winglang/wing/issues/6282#issuecomment-2068126152) compiled for `sim` platform.

Sure, yeah I think it would make sense for `cloud.Bucket` to be implemented in terms of `s3.Bucket`. I think there's still merit to having `cloud.Bucket` as its own class since e.g. Azure Blob Storage does not implement the exact contract of S3 AFAIK. This would be relevant if folks want to write apps that can be deployed to either AWS or Azure.
